VERSIONING.md

```diff
@@ -0,0 +1,90 @@
+      versioned in sync with the `go.opentelemetry.io/collector` module.
+    * Configuration structures should be considered part of the public API and backward
+      compatibility maintained through any changes made to configuration structures.
+         * Because configuration structures are typically instantiated through unmarshalling
+           a serialized representation of the structure, and not through structure literals,
+           additive changes to the set of exported fields in a configuration structure are
+           not considered to break backward compatibility.
+* Versioning of the associated [contrib
+  repository](https://github.com/open-telemetry/opentelemetry-collector-contrib) of
+  this project will be idiomatic of a Go project using [Go
+  modules](https://github.com/golang/go/wiki/Modules).
+    * [Semantic import
+      versioning](https://github.com/golang/go/wiki/Modules#semantic-import-versioning)
+      will be used.
+        * Versions will comply with [semver 2.0](https://semver.org/spec/v2.0.0.html).
+        * If a module is version `v2` or higher, the
+          major version of the module must be included as a `/vN` at the end of the
+          module paths used in `go.mod` files (e.g., `module
+          github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/v2`, `require
+          github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/v2 v2.0.1`) and in the
+          package import path (e.g., `import
+          "github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/v2"`). This includes
+          the paths used in `go get` commands (e.g., `go get
+          github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/v2@v2.0.1`.  Note there
+          is both a `/v2` and a `@v2.0.1` in that example. One way to think about
+          it is that the module name now includes the `/v2`, so include `/v2`
+          whenever you are using the module name).
+        * If a module is version `v0` or `v1`, do not include the major version
+          in either the module path or the import path.
+    * Configuration structures should be considered part of the public API and backward
+      compatibility maintained through any changes made to configuration structures.
+        * Because configuration structures are typically instantiated through unmarshalling
+          a serialized representation of the structure, and not through structure literals,
+          additive changes to the set of exported fields in a configuration structure are
+          not considered to break backward compatibility.
+    * Modules will be used to encapsulate receivers, processor, exporters,
+      extensions, and any other independent sets of related components.
+        * Experimental modules still under active development will be versioned at
+          `v0` to imply the stability guarantee defined by
```

Comment:
How would that work, given that the whole repository is tagged at once? For instance, once we are at v1.0.2, how would a component be marked as v0? As part of its module name in go.mod?

Reply:
Tags are created for each module and the tags do not need to be at identical versions.

<details>
<summary>
Tags for collector-contrib v0.28.0
</summary>

```sh
$ git tag | grep v0.28.0
exporter/alibabacloudlogserviceexporter/v0.28.0
exporter/awscloudwatchlogsexporter/v0.28.0
exporter/awsemfexporter/v0.28.0
exporter/awskinesisexporter/v0.28.0
exporter/awsprometheusremotewriteexporter/v0.28.0
exporter/awsxrayexporter/v0.28.0
exporter/azuremonitorexporter/v0.28.0
exporter/carbonexporter/v0.28.0
exporter/datadogexporter/v0.28.0
exporter/dynatraceexporter/v0.28.0
exporter/elasticexporter/v0.28.0
exporter/elasticsearchexporter/v0.28.0
exporter/f5cloudexporter/v0.28.0
exporter/googlecloudexporter/v0.28.0
exporter/honeycombexporter/v0.28.0
exporter/humioexporter/v0.28.0
exporter/influxdbexporter/v0.28.0
exporter/jaegerthrifthttpexporter/v0.28.0
exporter/loadbalancingexporter/v0.28.0
exporter/logzioexporter/v0.28.0
exporter/lokiexporter/v0.28.0
exporter/newrelicexporter/v0.28.0
exporter/sapmexporter/v0.28.0
exporter/sentryexporter/v0.28.0
exporter/signalfxexporter/v0.28.0
exporter/splunkhecexporter/v0.28.0
exporter/stackdriverexporter/v0.28.0
exporter/sumologicexporter/v0.28.0
exporter/tanzuobservabilityexporter/v0.28.0
exporter/uptraceexporter/v0.28.0
extension/fluentbitextension/v0.28.0
extension/httpforwarder/v0.28.0
extension/observer/ecsobserver/v0.28.0
extension/observer/hostobserver/v0.28.0
extension/observer/k8sobserver/v0.28.0
extension/observer/v0.28.0
extension/storage/v0.28.0
internal/aws/awsutil/v0.28.0
internal/aws/containerinsight/v0.28.0
internal/aws/k8s/v0.28.0
internal/aws/metrics/v0.28.0
internal/aws/xray/testdata/sampleapp/v0.28.0
internal/aws/xray/testdata/sampleserver/v0.28.0
internal/aws/xray/v0.28.0
internal/common/v0.28.0
internal/k8sconfig/v0.28.0
internal/kubelet/v0.28.0
internal/splunk/v0.28.0
internal/stanza/v0.28.0
internal/tools/v0.28.0
pkg/batchperresourceattr/v0.28.0
pkg/batchpersignal/v0.28.0
pkg/batchpertrace/v0.28.0
pkg/experimentalmetricmetadata/v0.28.0
processor/groupbyattrsprocessor/v0.28.0
processor/groupbytraceprocessor/v0.28.0
processor/k8sprocessor/v0.28.0
processor/metricsgenerationprocessor/v0.28.0
processor/metricstransformprocessor/v0.28.0
processor/resourcedetectionprocessor/v0.28.0
processor/routingprocessor/v0.28.0
processor/spanmetricsprocessor/v0.28.0
processor/tailsamplingprocessor/v0.28.0
receiver/awscontainerinsightreceiver/v0.28.0
receiver/awsecscontainermetricsreceiver/v0.28.0
receiver/awsxrayreceiver/v0.28.0
receiver/carbonreceiver/v0.28.0
receiver/collectdreceiver/v0.28.0
receiver/dockerstatsreceiver/v0.28.0
receiver/dotnetdiagnosticsreceiver/v0.28.0
receiver/filelogreceiver/v0.28.0
receiver/fluentforwardreceiver/v0.28.0
receiver/influxdbreceiver/v0.28.0
receiver/jmxreceiver/v0.28.0
receiver/k8sclusterreceiver/v0.28.0
receiver/kafkametricsreceiver/v0.28.0
receiver/kubeletstatsreceiver/v0.28.0
receiver/memcachedreceiver/v0.28.0
receiver/nginxreceiver/v0.28.0
receiver/prometheusexecreceiver/v0.28.0
receiver/receivercreator/v0.28.0
receiver/redisreceiver/v0.28.0
receiver/sapmreceiver/v0.28.0
receiver/signalfxreceiver/v0.28.0
receiver/simpleprometheusreceiver/examples/federation/prom-counter/v0.28.0
receiver/simpleprometheusreceiver/v0.28.0
receiver/splunkhecreceiver/v0.28.0
receiver/statsdreceiver/v0.28.0
receiver/syslogreceiver/v0.28.0
receiver/tcplogreceiver/v0.28.0
receiver/udplogreceiver/v0.28.0
receiver/wavefrontreceiver/v0.28.0
receiver/windowsperfcountersreceiver/v0.28.0
receiver/zookeeperreceiver/v0.28.0
testbed/mockdatareceivers/mockawsxrayreceiver/v0.28.0
testbed/v0.28.0
tracegen/v0.28.0
v0.28.0
```

</details>

<details>
<summary>
Tags at the same commit for opentelemetry-go v1.0.0-RC1/v0.21.0
</summary>

```
ced177b7 - (tag: v1.0.0-RC1, tag: trace/v1.0.0-RC1, tag: sdk/v1.0.0-RC1, tag: sdk/metric/v0.21.0, tag: sdk/export/metric/v0.21.0, tag: oteltest/v1.0.0-RC1, tag: metric/v0.21.0, tag: internal/metric/v0.21.0, tag: exporters/zipkin/v1.0.0-RC1, tag: exporters/trace/zipkin/v1.0.0-RC1, tag: exporters/trace/jaeger/v1.0.0-RC1, tag: exporters/stdout/stdouttrace/v1.0.0-RC1, tag: exporters/stdout/stdoutmetric/v0.21.0, tag: exporters/prometheus/v0.21.0, tag: exporters/otlp/otlptrace/v1.0.0-RC1, tag: exporters/otlp/otlptrace/otlptracehttp/v1.0.0-RC1, tag: exporters/otlp/otlptrace/otlptracegrpc/v1.0.0-RC1, tag: exporters/otlp/otlpmetric/v0.21.0, tag: exporters/otlp/otlpmetric/otlpmetricgrpc/v0.21.0, tag: exporters/metric/prometheus/v0.21.0, tag: exporters/jaeger/v1.0.0-RC1, tag: example/zipkin/v1.0.0-RC1, tag: example/prometheus/v0.21.0, tag: example/passthrough/v1.0.0-RC1, tag: example/otel-collector/v1.0.0-RC1, tag: example/opencensus/v0.21.0, tag: example/namedtracer/v1.0.0-RC1, tag: example/jaeger/v1.0.0-RC1, tag: bridge/opentracing/v0.21.0, tag: bridge/opencensus/v0.21.0) Pre-release 1.0.0-RC1 (#2013) (6 days ago) <Anthony Mirabella>
```

</details>

Reply:
I can't understand how people are able to keep their sanity with multi-module mono-repos...

Reply:
It's all held together with bash scripts and duct tape.  Much like the rest of the internet! :)